tests/system/libraries/NvdaLib.py

```diff
@@ -0,0 +1,243 @@
+# For more details see: https://www.gnu.org/licenses/gpl-2.0.html
+
+"""This file provides robot library functions for NVDA system tests.
+It contains helper methods for system tests, most specifically related to NVDA
+- setup config,
+- starting
+- quiting
+- config cleanup
+This is in contrast with the `SystemTestSpy/speechSpy*.py files,
+which provide library functions related to monitoring NVDA and asserting NVDA output.
+"""
+# imported methods start with underscore (_) so they don't get imported into robot files as keywords
+from os.path import join as _pJoin, abspath as _abspath, expandvars as _expandvars
+import tempfile as _tempFile
+from typing import Optional
+
+from robotremoteserver import (
+	test_remote_server as _testRemoteServer,
+	stop_remote_server as _stopRemoteServer,
+)
+from SystemTestSpy import (
+	_blockUntilConditionMet,
+	_getLib,
+	_nvdaSpyAlias,
+	configManager
+)
+
+# Imported for type information
+from robot.libraries.BuiltIn import BuiltIn
+from robot.libraries.OperatingSystem import OperatingSystem as _OpSysLib
+from robot.libraries.Process import Process as _Process
+from robot.libraries.Remote import Remote as _Remote
+
+builtIn: BuiltIn = BuiltIn()
+opSys: _OpSysLib = _getLib('OperatingSystem')
+process: _Process = _getLib('Process')
+
+
+class _NvdaLocationData:
+
+	def __init__(self):
+		# robot is expected to be run from the NVDA repo root directory. We want all repo specific
+		# paths to be relative to this. This would allow us to change where it is run from if we decided to.
+		self.repoRoot = _abspath("./")
+		self.stagingDir = _tempFile.gettempdir()
+		opSys.directory_should_exist(self.stagingDir)
+
+		self.whichNVDA = builtIn.get_variable_value("${whichNVDA}", "source")
+		if self.whichNVDA == "source":
+			self._runNVDAFilePath = _pJoin(self.repoRoot, "source/nvda.pyw")
+			self.baseNVDACommandline = f"pyw -3.7-32 {self._runNVDAFilePath}"
+		elif self.whichNVDA == "installed":
+			self._runNVDAFilePath = _pJoin(_expandvars('%PROGRAMFILES%'), 'nvda', 'nvda.exe')
+			self.baseNVDACommandline = f'"{str(self._runNVDAFilePath)}"'
+		else:
+			raise AssertionError("RobotFramework should be run with argument: '-v whichNVDA [source|installed]'")
+
+		self.profileDir = _pJoin(self.stagingDir, "nvdaProfile")
+		self.logPath = _pJoin(self.profileDir, 'nvda.log')
+		self.preservedLogsDir = _pJoin(
+			builtIn.get_variable_value("${OUTPUT DIR}"),
+			"nvdaTestRunLogs"
+		)
+
+	def ensurePathsExist(self):
+		opSys.file_should_exist(self._runNVDAFilePath, "Unable to start NVDA unless path exists.")
+		opSys.create_directory(self.profileDir)
+		opSys.create_directory(self.preservedLogsDir)
+
+
+_locations = _NvdaLocationData()
+
+
+class NvdaLib:
+	"""Robot Framework library for interacting with NVDA.
+	Notable:
+	- NvdaLib.nvdaSpy is a library instance for getting speech and other information out of NVDA
+	"""
+	def __init__(self):
+		self.nvdaSpy = None  #: Optional[SystemTestSpy.speechSpyGlobalPlugin.NVDASpyLib]

```

Comment:
Why is this annotated like this?

Reply:
So that we don't have to import that module just for the type. That module depends on other bits of NVDA so it gets messy to import it.